tests/test_sync.py

```diff
@@ -137,17 +157,123 @@ def test_copy_part_gs_to_s3(self):
+        }
+
+        collection_key = "{}/{}".format(COLLECTION_PREFIX, get_collection_fqid())
+        collection_blob = self.s3_bucket.Object(collection_key)
+        collection_blob.put(Body=json.dumps(collection_data).encode())
+        self.assertFalse(sync.dependencies_exist(Replica.aws, Replica.aws, collection_key))
+
+        bundle_key = "{}/{}".format(BUNDLE_PREFIX, BundleFQID(uuid=bundle_uuid, version=bundle_version))
+        bundle_blob = self.s3_bucket.Object(bundle_key)
+        bundle_blob.put(Body=json.dumps(bundle_data).encode())
+
+        self.assertFalse(sync.dependencies_exist(Replica.aws, Replica.aws, collection_key))
+        self.assertFalse(sync.dependencies_exist(Replica.aws, Replica.aws, bundle_key))
+
+        file_key = "{}/{}".format(FILE_PREFIX, FileFQID(uuid=file_uuid, version=file_version))
+        file_blob = self.s3_bucket.Object(file_key)
+        file_blob.put(Body=json.dumps(file_data).encode())
+
+        @eventually(timeout=8, interval=1, errors={Exception})
+        def check_file_revdeps():
+            self.assertTrue(sync.dependencies_exist(Replica.aws, Replica.aws, collection_key))
+            self.assertTrue(sync.dependencies_exist(Replica.aws, Replica.aws, bundle_key))
+            self.assertFalse(sync.dependencies_exist(Replica.aws, Replica.aws, file_key))
+        check_file_revdeps()
+
+        blob_key = compose_blob_key(file_data)
+        blob_blob = self.s3_bucket.Object(blob_key)
+        blob_blob.put(Body=b"sync_test")
+
+        @eventually(timeout=8, interval=1, errors={Exception})
+        def check_blob_revdeps():
+            self.assertTrue(sync.dependencies_exist(Replica.aws, Replica.aws, collection_key))
+            self.assertTrue(sync.dependencies_exist(Replica.aws, Replica.aws, bundle_key))
+            self.assertTrue(sync.dependencies_exist(Replica.aws, Replica.aws, file_key))
+        check_blob_revdeps()
+
+# TODO: (akislyuk) integration test of SQS fault injection, SFN fault injection
+
+@testmode.integration
+class TestSyncDaemon(unittest.TestCase, TestSync):
+    def setUp(self):
+        dss.Config.set_config(dss.BucketConfig.TEST)
+        self.gs_bucket_name, self.s3_bucket_name = dss.Config.get_gs_bucket(), dss.Config.get_s3_bucket()
+        self.logger = logging.getLogger(__name__)
+        self.gs = Config.get_native_handle(Replica.gcp)
+        self.gs_bucket = self.gs.bucket(self.gs_bucket_name)
+        self.s3 = boto3.resource("s3")
+        self.s3_bucket = self.s3.Bucket(self.s3_bucket_name)
+
+    def test_sync_small_blob(self):
+        "Tests oneshot blob syncing."
+        self._sync_blob()
+
+    @unittest.skipUnless(json.loads(os.environ.get("DSS_TEST_LARGE_FILE_SYNC", "false")),
+                         "Skipping large file sync test. Set DSS_TEST_LARGE_FILE_SYNC=1 to run it")
+    def test_sync_large_blob(self):
+        "Tests multipart blob syncing."
+        self._sync_blob(payload_size=max(sync.part_size.values()) + 1)
```

Comment:
add tests around the multipart copy threshold. -1, 0, +1

Reply:
Added.